bin/user_config_converter.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python3
+
+from enum import Enum, auto
+import importlib
+import json
+import os
+from pathlib import Path
+import sys
+from typing import Type
+
+from fusion_engine_client.messages.configuration import PlatformStorageDataMessage
+from fusion_engine_client.parsers.decoder import FusionEngineDecoder
+
+# Add the parent directory to the search path to enable p1_runner package imports when not installed in Python.
+repo_root = Path(os.path.dirname(__file__)).parent.absolute()
+sys.path.append(str(repo_root))
+
+# Example version imported just to help with type checking
+from user_config_loaders.platform_1.version_7_1.user_config_loader.user_config_loader import UserConfig as UserConfigTypingClass
+
+from p1_runner import trace as logging
+from p1_runner.argument_parser import ArgumentParser
+
+logger = logging.getLogger('point_one.user_config_converter')
+
+
+class ConversionDirection(Enum):
+    TO_JSON = auto()
+    TO_BINARY = auto()
+
+
+def main():
+    if getattr(sys, 'frozen', False):
+        execute_command = os.path.basename(sys.executable)
+    else:
+        execute_command = os.path.basename(sys.executable)
+        if execute_command.startswith('python'):
+            execute_command += ' ' + os.path.basename(__file__)
+
+    parser = ArgumentParser(
+        usage='%s COMMAND [OPTIONS]... IN_FILE OUT_FILE' % execute_command,
+        description='Convert user configuration files between binary and JSON representations.')
+
+    parser.add_argument('-v', '--verbose', action='count', default=0,
+                        help="Print verbose/trace debugging messages. May be specified multiple times to increase "
+                             "verbosity.")
+
+    parser.add_argument(
+        'in_file',
+        type=Path,
+        help="The path to the file to load. Must have extension '*.json' or '*.p1log'.")
+    parser.add_argument(
+        'out_file',
+        type=Path,
+        help="The path to write the converter file to. If this is a directory, the filename will be based on the input file with new extension. For example /in_dir/bar.json -> /out_file/bar.p1log.")
```

Comment:
Would be nice to make this argument optional. If omitted, we'll just use the path and stem from the input file and change the extension.